### PR TITLE
Clean up MSBuild files

### PR DESCRIPTION
**Pull request by JunielKatarn**
- Remove redundant property and flag declarations.
- Fix Visual Studio IDE compatibility issue with x86 (Win32) builds.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/2664)

### CODE REVIEW

File: vnext/ReactCommon/ReactCommon.vcxproj [hunk 1]
```diff
@@ -95,7 +86,7 @@
   </ItemDefinitionGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
-    <Import Project="$(SolutionDir)\packages\boost.1.68.0.0\build\boost.targets" Condition="Exists('$(SolutionDir)\packages\boost.1.68.0.0\build\boost.targets')" />
```
Comment (hansenyy) [inline vnext/ReactCommon/ReactCommon.vcxproj]: >r)\pa [](start = 33, length = 5)

Is this change intentional?

Comment (JunielKatarn) [reply vnext/ReactCommon/ReactCommon.vcxproj]: Yes. MSBuild ensures `SolutionDir` always has a trailing slash. Adding one is redundant.

File: vnext/Desktop.DLL/React.Windows.Desktop.DLL.vcxproj [hunk 2]
```diff
@@ -1,14 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <ItemGroup Label="ProjectConfigurations">
-    <ProjectConfiguration Include="Debug|x86">
```
Comment (aeulitz) [inline vnext/Desktop.DLL/React.Windows.Desktop.DLL.vcxproj]: Is that being replaced with another config?

File: vnext/.editorconfig [hunk 3]
```diff
@@ -14,7 +14,7 @@ indent_size = 2
 end_of_line = crlf
 
 # Xml project files
-[*.{csproj,vcxproj,pssproj,vcxproj.filters,targets,props}]
+[*.{csproj,vcxproj,pssproj,vcxproj.filters,vcxitems,targets,props}]
```
Comment (NikoAri) [inline vnext/.editorconfig]: >vcxitems [](start = 43, length = 8)

(nit) can we alphabenitze those?

File: vnext/Desktop.DLL/React.Windows.Desktop.DLL.vcxproj (hunk 2 above)
Comment (JunielKatarn) [reply vnext/Desktop.DLL/React.Windows.Desktop.DLL.vcxproj]: The actual config is `Win32`. This was a hack to help the output folders path use `x86` instead.
This caused problems when running tests or manipulating nuget packages within the IDE.

This has been fixed, and this hack is no longer needed.

File: vnext/Directory.Build.props [hunk 4]
```diff
@@ -8,33 +8,30 @@
...
-    <!-- Windows Desktop -->
     <IntDir>$(BaseIntDir)\$(ProjectName)\</IntDir>
     <OutDir>$(BaseOutDir)\$(ProjectName)\</OutDir>
-  </PropertyGroup>
 
-  <PropertyGroup>
-    <FollyDir>$(ReactNativeWindowsDir)node_modules\react-native\folly</FollyDir>
-    <ReactNativeDir>$(ReactNativeWindowsDir)node_modules\react-native</ReactNativeDir>
-    <YogaDir>$(ReactNativeWindowsDir)node_modules\react-native\ReactCommon\yoga</YogaDir>
+    <GeneratedFilesDir>$(IntDir)Generated Files\</GeneratedFilesDir>
```
Comment (NikoAri) [inline vnext/Directory.Build.props]: >$(IntDir) [](start = 23, length = 9)

are directory related variables suppose to contain backslash character '\' at the end?
BaseIntDir doesn't have backslash, but IntDir does.
Please make them consistent.

Comment (JunielKatarn) [reply vnext/Directory.Build.props]: The inconsistency comes from MSBuild conventions themselves.
In particular, `SolutionDir` is enforced to have a trailing slash.
Other variables, not necessarily.
(If curious, you can look at the VCTargets path where these variables are defined).

File: vnext/.editorconfig (hunk 3 above)
Comment (JunielKatarn) [reply vnext/.editorconfig]: Sorry! Missed this comment.
Will sort these in a subsequent PR.

**Comment by NikoAri** [comment]
>     

remove this line here and in other files

---
Refers to: vnext/Desktop.UnitTests/React.Windows.Desktop.UnitTests.vcxproj:12 in 1aec2c9. [](commit_id = 1aec2c9b6d78d0c128deb8300aedee16c42976e5, deletion_comment = False)

File: vnext/ReactWindows.sln [hunk 5]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.27310.2036
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.29009.5
```
Comment (aeulitz) [inline vnext/ReactWindows.sln]: >16.0.29009.5 [](start = 22, length = 12)

Has this any ramifications on the VS version needed to open the solution?

File: vnext/PropertySheets/React.Cpp.props [hunk 6]
```diff
@@ -43,7 +42,7 @@
     </Link>
   </ItemDefinitionGroup>
 
-  <ItemDefinitionGroup Condition="'$(Platform)' == 'x64' OR '$(Platform)' == 'x86'">
+  <ItemDefinitionGroup Condition="'$(Platform)' == 'x64' OR '$(Platform)' == 'x86' OR '$(Platform)' == 'Win32'">
```
Comment (aeulitz) [inline vnext/PropertySheets/React.Cpp.props]: >'x86' OR '$(Platform)' == 'Win32' [](start = 77, length = 33)

I'm confused - what is the name of the platform now, 'x86' or 'Win32'?

Comment (JunielKatarn) [reply vnext/PropertySheets/React.Cpp.props]: For C++ projects, `Win32` was the value used to refer to the  32 bit `ix86` architecture.
After `x64` was introduced, the MSVC team realized `x86` was a better alternativer, but for some workflows (like Visual Studio IDE project loading), `Win32` is still the value used by the build system.

In short, building the same VCXPROJ with either `/p:Platform=Win32` or `/p:Platform=x86` yields the same binary artifacts, but the paths for `$(IntDir)` and `$(OutDir)` may differ.

These predicates ensure that whether building within the IDE or from command line, we get the same behavior.

File: vnext/ReactWindows.sln (hunk 5 above)
Comment (JunielKatarn) [reply vnext/ReactWindows.sln]: No. The solution can actualy be opened by VS 2017. I chose to update the value to suggest that VS2019 is preferred.

File: vnext/PropertySheets/React.Cpp.props (hunk 6 above)
Comment (aeulitz) [reply vnext/PropertySheets/React.Cpp.props]: Pardon the somewhat facetious question. I understand what that means, I'm questioning the value of a choice without distinction. Why not universally refer to the platform as Win32?

---
In reply to: [296922183](https://github.com/microsoft/react-native-windows/pull/2664#discussion_r296922183) [](ancestors = 296922183)

Comment (JunielKatarn) [reply vnext/PropertySheets/React.Cpp.props]: Discussed offline.
$(Platform) can be set as either `Win32` or `x86`. The output paths will contain `x86` irregardless.
